rust/src/database/database.rs

```diff
@@ -179,43 +179,35 @@ impl Database {
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
     pub(super) async fn run_on_any_replica<F, P, R>(&self, task: F) -> Result<R>
     where
-        F: Fn(ServerDatabase, ServerConnection, bool) -> P,
+        F: Fn(ServerDatabase, ServerConnection) -> P,
         P: Future<Output = Result<R>>,
     {
-        let mut is_first_run = true;
```

Comment:
`Session::transaction()` was the only place that used this flag. It is no longer necessary.